docs/CustomFunctions.md

```diff
@@ -0,0 +1,132 @@
+
+```
+return_type function_name(type $argname)
+```
+
+The signature indicates the the valid types for the arugments. If an invalid type is passed in for an argument an error will occur.
+
+When math related functions are performed the end result _must_ be able to fit within a C# [long](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types#characteristics-of-the-integral-types) value. If this cannot happen then a mathmatical error will occur.
+
+Please see the [specification](https://jmespath.org/specification.html#built-in-functions) for more details.
+
+## Mathmatical Functions
+
+### add
+
+```
+number add(number $left, number $right)
+```
+
+Returns the result of adding the left argument from the right.
+
+Examples:
+| Given                       | Expression       | Result |
+|-----------------------------|------------------|--------|
+| n/a                         | add(`10`, `10`)  | 20     |
+| {"left" : 40, "right" : 50} | add(left, right) | 90     |
+| {"left" : 0, "right" : 50}  | add(left, right) | 50     |
+
+### divide
+
+```
+number divide(number $left, number $right)
+```
+
+Returns the result of multiplying the left argument with the right.
+
+Examples:
+| Given                       | Expression          | Result |
+|-----------------------------|---------------------|--------|
+| n/a                         | divide(`10`, `10`)  | 1      |
+| {"left" : 40, "right" : 50} | divide(left, right) | 0.8    |
+| {"left" : 0, "right" : 50}  | divide(left, right) | 0      |
+
+### multiply
+
+```
+number multiply(number $left, number $right)
+```
+
+Returns the result of multiplying the left argument with the right.
+
+Examples:
+| Given                       | Expression            | Result |
+|-----------------------------|-----------------------|--------|
+| n/a                         | multiply(`10`, `10`)  | 100    |
+| {"left" : 40, "right" : 50} | multiply(left, right) | 2000   |
+| {"left" : 0, "right" : 50}  | multiply(left, right) | 0      |
+
+
+### pow
+
+```
+number pow(number $left, number $right)
+```
+
+Returns the result of raising the left argument to the power of the right.
+
+Examples:
+| Given                         | Expression       | Result                     |
+|-------------------------------|------------------|----------------------------|
+| n/a                           | pow(`10`, `10`)  | 10000000000                |
+| {"left" : 40, "right" : 50}   | pow(left, right) | mathmatic error : overflow |
+| {"left" : 0, "right" : 50}    | pow(left, right) | 0                          |
+| {"left" : 100, "right" : 0.5} | pow(left, right) | 10                         |
+
+### subtract
+
+```
+number subtract(number $left, number $right)
+```
+
+Returns the result of subtracting the left argument from the right.
```

Comment:
Based on the table below it looks like we are actually subtracting the right argument **from** the left.